### PR TITLE
feat: upgrade to spectral 6 and convert to monorepo

### PR DESCRIPTION
**Pull request by dpopp07**
This commit refactors the project into a monorepo and separates the project
into two NPM packages. One is the existing ibm-openapi-validator package and
one is the new package for our custom Spectral ruleset. As part of this change,
we upgraded to Spectral 6 which contained a number of breaking changes.

There are some incompatible changes for this tool (not many) as a result. These
are described in the migration guide. Since a number of these files were being
moved around already, I took the opportunity to rename a bunch of files to match
our kebab-case convention of file naming.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ script:
 deploy:
 - provider: script
   skip_cleanup: true
-  script: npx semantic-release
+  # script: npx -p @qiwi/semrel-toolkit multi-semrel
```
Comment (dpopp07) [inline .travis.yml:13]: Will uncomment this before merging, just didn't want to make any unintentional moves before everything is in place

File: packages/ruleset/src/functions/required-property.js [hunk 2]
```diff
@@ -34,7 +33,7 @@ function checkRequiredProperties(schema, path) {
         let message;
         if (schema.allOf) {
           message = `Required property, ${requiredPropName}, must be defined in at least one of the allOf schemas`;
-        } else if (schema.anyOf || schema.allOf) {
+        } else if (schema.anyOf || schema.oneOf) {
```
Comment (dpopp07) [inline packages/ruleset/src/functions/required-property.js:36]: Caught a bug with the new tests 🙂

File: packages/ruleset/src/functions/string-boundary.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
-module.exports = function(schema, _opts, paths) {
-  const rootPath = paths.target !== void 0 ? paths.target : paths.given;
```
Comment (dpopp07) [inline packages/ruleset/src/functions/string-boundary.js:2]: This line is no longer relevant with the new spectral - it is removed from a number of files

File: packages/ruleset/test/utils/root-document.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
+module.exports = {
+  openapi: '3.0.0',
```
Comment (dpopp07) [inline packages/ruleset/test/utils/root-document.js:2]: This is a clean API definition that I use for the tests. Easier to edit and make one thing go wrong than it is to write new definitions a bunch

File: packages/validator/src/.defaultsForValidator.js [hunk 5]
```diff
@@ -98,29 +98,6 @@ const defaults = {
...
-      'oas2-host-trailing-slash': "warning",
-      'oas2-valid-definition-example': "warning",
-      'oas2-anyOf': "warning",
-      'oas2-oneOf': "warning",
-      'oas3-api-servers': "warning",
-      'oas3-examples-value-or-externalValue': "warning",
-      'oas3-server-trailing-slash': "warning",
-      'oas3-valid-oas-content-example': "warning",
-      'oas3-valid-schema-example': "warning"
-    }
```
Comment (dpopp07) [inline packages/validator/src/.defaultsForValidator.js:123]: I decided we shouldn't support configuring Spectral rules from the `.validaterc` file anymore. It's just a confusing way to do it and it's tough to merge these rules in. If that is too heavy handed, let me know

File: packages/validator/src/cli-validator/run-validator.js [hunk 6]
```diff
@@ -261,6 +254,13 @@ const processInput = async function(program) {
       if (debug) {
         console.log(err.stack);
       }
+      // this check can be removed once we support spectral overrides
+      if (err.message.startsWith('Document must have some source assigned.')) {
+        console.log(
+          'This error likely occurred because Spectral `exceptions` are deprecated and `overrides` are not yet supported.\n' +
+            'Remove these fields from your Spectral config file to proceed.'
+        );
```
Comment (dpopp07) [inline packages/validator/src/cli-validator/run-validator.js:262]: Due to some bizarre behavior in Spectral, we can't support overrides yet. That will have to be a follow on change. This gives users a better error message than Spectral does

File: packages/validator/src/lib/index.js [hunk 7]
```diff
@@ -17,20 +18,17 @@ module.exports = async function(
   // or the default ruleset
   let configObject;
   let spectralResults;
-  const spectral = new Spectral({
-    computeFingerprint: dedupFunction
-  });
```
Comment (dpopp07) [inline packages/validator/src/lib/index.js:22]: We were duplicating too much setup code, I moved all of this into the `setup` function that is called later

File: packages/validator/test/cli-validator/alternate-spectral-configs/extends-default.yaml [hunk 8]
```diff
@@ -1,4 +1,4 @@
-extends: ibm:oas
+extends: '@dpopp07/test-ibm-oas-ruleset'
```
Comment (dpopp07) [inline packages/validator/test/cli-validator/alternate-spectral-configs/extends-default.yaml:1]: I'll change all instances of this before merging

File: packages/validator/test/cli-validator/mock-files/oas3/testoneof.yaml [hunk 9]
```diff
@@ -1,4 +1,6 @@
 openapi: 3.0.0
+servers:
+  - url: https://oneoftester.com/api
```
Comment (dpopp07) [inline packages/validator/test/cli-validator/mock-files/oas3/testoneof.yaml:3]: Spectral throws the craziest exception if an API def is missing a servers field lol. I plan to open an issue about it at some point

**Review by padamstx — APPROVED** [review]
Looks great to me!
Other than the version # issue I mentioned on slack, I can't find anything else to complain about!!! :)

**Review by hudlow — APPROVED** [review]
Looks good as far as I was able to understand it.

**Pull request merged**